### PR TITLE
go.mod: remove `replace inet.af/tcpproxy => github.com/inetaf/tcpproxy`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
By copying <https://github.com/inetaf/tcpproxy/blob/91f861402626/tcpproxy.go> to `pkg/tcpproxy/tcpproxy.go`.

### CODE REVIEW

File: pkg/tcpproxy/tcpproxy.go [hunk 1]
```diff
@@ -81,16 +83,6 @@ type Proxy struct {
 	ListenFunc func(net, laddr string) (net.Listener, error)
 }
 
-// Matcher reports whether hostname matches the Matcher's criteria.
-type Matcher func(ctx context.Context, hostname string) bool
-
-// equals is a trivial Matcher that implements string equality.
-func equals(want string) Matcher {
```
Comment (AkihiroSuda) [inline pkg/tcpproxy/tcpproxy.go:88]: (This unused function had to be removed to silence the linter)

**Comment by AkihiroSuda** [comment]
Test failure is irrelevant https://github.com/containers/gvisor-tap-vsock/actions/runs/11388213298/job/31684676909?pr=409

```
E: Failed to fetch mirror+file:/etc/apt/apt-mirrors.txt/pool/main/e/edk2/ovmf_2022.02-3ubuntu0.22.04.2_all.deb  404  Not Found [IP: 52.252.75.106 80]
```

**Comment by evidolob** [comment]
@cfergeau I could confirm that this PR has no issue https://github.com/containers/gvisor-tap-vsock/issues/387 
It seems that everything works fine

**Comment by cfergeau** [comment]
> @cfergeau I can confirm that this PR has no issue #387 It seems that everything works fine

Ah great, thanks for checking!

**Review by cfergeau — APPROVED** [review]

**Pull request merged**